src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -320,32 +371,50 @@ internal static IEnumerable<Transaction> Order(IDictionary<AddressAsKey, Transac
                 // A -> N0_P3, N1_P1, N1_P0, N3_P5...
                 // B -> N4_P4, N5_P3, N6_P3...
                 // We construct [N4_P4 (B), N0_P3 (A)] in sorted order by priority
-                DictionarySortedSet<Transaction, IEnumerator<Transaction>> transactions = new(comparerWithIdentity);
+                DictionarySortedSet<Transaction, (IEnumerator<Transaction> enumerator, long gasUsed, int blobCount)> transactions = new(comparerWithIdentity);
 
                 for (int i = 0; i < bySenderEnumerators.Length; i++)
                 {
                     IEnumerator<Transaction> enumerator = bySenderEnumerators[i];
                     if (enumerator.MoveNext())
                     {
-                        transactions.Add(enumerator.Current!, enumerator);
+                        Transaction current = enumerator.Current!;
+                        transactions.Add(current, (enumerator, 0, 0));
                     }
                 }
 
                 // while there are still unreturned transactions
                 while (transactions.Count > 0)
                 {
                     // we take first transaction from sorting order, on first call: N4_P4 from B
-                    (Transaction tx, IEnumerator<Transaction> enumerator) = transactions.Min;
+                    (Transaction addedTx, (IEnumerator<Transaction> enumerator, long gasUsed, int blobCount)) = transactions.Min;
+
+                    transactions.Remove(addedTx);
+                    if (filter is not null && !filter.Invoke(addedTx))
+                    {
+                        // Cannot be added, so stop adding more txs from this sender
+                        continue;
+                    }
```

Comment:
We now invoke this filter 3x per transaction - seems wasteful?

Reply:
Changed to only run once